be/dtm0_log.h

```diff
@@ -277,13 +277,21 @@ M0_INTERNAL void m0_be_dtm0_log_fini(struct m0_be_dtm0_log *log);
  * Free the memory allocated by m0_be_dtm0_log_alloc
  *
  * @pre m0_be_dtm0_log->dl_is_persistent needs to be false.
- * @post *log is set to NULL.
+ * @post None
  *
  * @param log Pointer to a log structure that has been previously allocated.
  *
  * @return None
  */
-M0_INTERNAL void m0_be_dtm0_log_free(struct m0_be_dtm0_log **log);
+M0_INTERNAL void m0_be_dtm0_log_free(struct m0_be_dtm0_log *log);
+
+/** Frees memory and zeroes the pointer. */
+#define m0_be_dtm0_log_free0(pptr)            \
```

Comment:
Personally, I don't like this macro.
1. No need to have such a macro. It can be replaced just with the actual code.
2. No need to pass "pointer to a pointer" to this macro. Macro is not like a function. The parameter can be used and modified directly. So, just pass the "struct m0_be_dtm0_log *log" type of variable.

Reply:
This was done as exact clone of m0_free0:

```
/** Frees memory and unsets the pointer. */
#define m0_free0(pptr)                        \
        do {                                  \
                typeof(pptr) __pptr = (pptr); \
                m0_free(*__pptr);             \
                *__pptr = NULL;               \
        } while (0)
```

Do you still think we should modify it?  Or keep similar?

Reply:
It's fine to keep it.